### PR TITLE
[DevAsan] Add an option to enable/disable leak detection

### PR DESCRIPTION
**Pull request by yingcong-wu**
It's better that we have an such option when we need to disable the leak detection.

### CODE REVIEW

File: source/loader/layers/sanitizer/asan_interceptor.hpp [hunk 1]
```diff
@@ -291,6 +291,7 @@ class SanitizerInterceptor {
                                   std::shared_ptr<DeviceInfo> &DeviceInfo);
 
   private:
+    AsanOptions m_Options;
```
Comment (AllanZyne) [inline source/loader/layers/sanitizer/asan_interceptor.hpp:294]: unnecessary changes

**Review by pbalcer — APPROVED** [review]

**Comment by yingcong-wu** [comment]
Convert this to draft in order to fast-track https://github.com/oneapi-src/unified-runtime/pull/2232 .

**Comment by yingcong-wu** [comment]
CI fails with apt package installation. Could we restart the CI?

**Review by pbalcer — APPROVED** [review]

**Review by AllanZyne — APPROVED** [review]
